### ISSUE TITLE
Allow for different interp options for energies in EnergyDispersion

### ISSUE THREAD

**Issue by adonath** [gammapy/gammapy#2457]
Currently our `EnergyDispersion` object is hard-coded to logarithmically spaced energy bins. On init the `MapAxis` object is created with `interp="log"`, independent of what the actual spacing of the energy bins in `e_reco` and `e_true` is. @registerrier mentioned in private this might be an issue when sometimes e.g. in X-ray astronomy RMF matrices are stored with linearly spaced energy bins. This is a reminder issue to fix this.

**Comment by cdeil** [comment]
[EnergyDispersion](https://docs.gammapy.org/0.14/api/gammapy.irf.EnergyDispersion.html) is basically a matrix, and it supports arbitrary energy binnings. 

What's the issue? If an axis is involved, it's `interp` isn't used / irrelevant, no? Can you give a code example / test case that shows the issue?

**Comment by adonath** [comment]
Sorry, the explanation in the initial post was a bit short.

The `EnergyDispersion` object currently holds the information about the reconstructed energy axis in `EnergyDispersion.e_reco`. If the edisp is applied to a map, then the new map axis is created from the `EnergyDispersion.e_reco` axis. Currently the `interp` type is hard-coded in `EnergyDispersion.__init__` (see [here](https://github.com/gammapy/gammapy/blob/master/gammapy/irf/energy_dispersion.py#L76)). So the resulting map axis is always defined with `interp="log"`. Here is a minimal example:
```
from gammapy.maps import Map, MapAxis
from gammapy.irf import EnergyDispersion
from astropy import units as u

e_true_axis = MapAxis.from_bounds(1, 10, nbin=10, unit="TeV", name="energy")

map_e_true = Map.create(width=10, axes=[e_true_axis])
e_reco = [1, 2, 4, 8, 10] * u.TeV
edisp = EnergyDispersion.from_diagonal_response(e_reco=e_reco, e_true=axis.edges)
map_e_reco = map_e_true.apply_edisp(edisp)

print(map_e_reco.geom.axes[0])
```
I guess there are also other ways to fix this behaviour, but I think in general we should just instantiate the `EnergyDisperision` with `MapAxis` objects as well, so that this information is kept and can be defined by the user, if needed.

**Comment by cdeil** [comment]
Hmmm ... I'm not sure offering the option of `lin` energy axis is a good idea. It doubles the effort for testing (e.g. for integration and science results), docs (how it works, when to use which) and has a pretty high cost for basically no gain (or negative gain, inviting user error).

Suggest to only support `log` interpolation and `log-log` integration for energy axes.

There is the manpower / time issue for v1.0, but even apart from that, it's not clear to me that offering `lin` energy axis is a good idea.

**Comment by cdeil** [comment]
Even if you use `MapAxis` for energy axes in IRFs, you don't have to use and support both interp / integrate modes for it. It's OK to have a more generic data structure (MapAxis), but to only use a subset of its features for a specific application (energy axis), no?

**Comment by adonath** [comment]
I agree it's certainly not a priority. I encountered the issue, because we (accidentally) created energy axis with `interp="lin"` in some tests (fixed in #2456). @registerrier then mentioned, that RMF matrices in X-ray astronomy are sometimes in linear spacing and @facero might be using those for some prototype X-ray analysis with Gammapy.

**Comment by cdeil** [comment]
> RMF matrices in X-ray astronomy are sometimes in linear spacing

I don't think that's a thing. As far as I know, an RMF matrix is a matrix, with some energy binning.
There is no "interp" property attached to it, that would trigger different interp / integrate behaviour.
I've never seen such an option e.g. when doing a spectral analysis in Sherpa.

**Comment by adonath** [comment]
For clarification: indeed the matrix is a matrix and there is now different behaviour for integration or interpolation, however the energy axis information is copied over in `Map.apply_edisp()`, so the resulting map object has a different axis scaling. This issue is resolved by #3185, anyway, as the `EDispKernel` now takes `MapAxis` objects as well.
